src/test/java/com/saucelabs/jenkins/pipeline/SauceStepTest.java

```diff
@@ -0,0 +1,118 @@
+                "echo 'ACCESS_KEY=' + env.SAUCE_ACCESS_KEY\n" +
+
+                "}}",
+            true
+        ));
+        WorkflowRun run = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("USERNAME=fakeuser", run);
+        r.assertLogContains("ACCESS_KEY=fakekey", run);
+    }
+
+    @Test
+    @Ignore("Mock fails to stick when running with agents")
+    public void sauceConnectTest() throws Exception {
+        String credentialsId = SauceCredentials.migrateToCredentials("fakeuser", "fakekey", "unittest");
+
+        SauceConnectFourManager sauceConnectFourManager = Mockito.mock(SauceConnectFourManager.class);
+
+        storeDummyManager(sauceConnectFourManager);
+        // stubbing appears before the actual execution
+        Mockito.when(sauceConnectFourManager.openConnection(
+            Mockito.anyString(),
+            Mockito.anyString(),
+            Mockito.anyInt(),
+            Mockito.any(File.class),
+            Mockito.anyString(),
+            Mockito.any(PrintStream.class),
+            Mockito.eq(true),
+            Mockito.anyString()
+        )).thenReturn(null);
+
+        DumbSlave s = r.createSlave();
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+            "node('" + s.getNodeName() + "') { sauce('" + credentialsId + "') { sauceconnect(useGeneratedTunnelIdentifier: true, verboseLogging: true) { \n" +
+                "echo 'USERNAME=' + env.SAUCE_USERNAME\n" +
+                "echo 'ACCESS_KEY=' + env.SAUCE_ACCESS_KEY\n" +
+            "}}}",
+            true
+        ));
+        WorkflowRun run = r.assertBuildStatusSuccess(p.scheduleBuild2(0));
+        r.assertLogContains("USERNAME=fakeuser", run);
+        r.assertLogContains("ACCESS_KEY=fakekey", run);
+
+        Mockito.verify(sauceConnectFourManager).openConnection(
+            Mockito.eq("fakeuser"),
+            Mockito.eq("fakekey"),
+            Mockito.anyInt(),
+            Mockito.any(File.class),
+            Mockito.anyString(),
+            Mockito.any(PrintStream.class),
+            Mockito.eq(true),
+            Mockito.anyString()
+        );
+    }
+
+    @Test
+    public void sauceConnectWithoutSauceTest() throws Exception {
+        DumbSlave s = r.createSlave();
+        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
+        p.setDefinition(new CpsFlowDefinition(
+            "node('" + s.getNodeName() + "') { sauceconnect(useGeneratedTunnelIdentifier: true, verboseLogging: true) { \n" +
+                "echo 'USERNAME=' + env.SAUCE_USERNAME\n" +
+                "echo 'ACCESS_KEY=' + env.SAUCE_ACCESS_KEY\n" +
+                "}}",
+            true
+        ));
+        WorkflowRun run = p.scheduleBuild2(0).get();
+        r.assertBuildStatus(Result.FAILURE, run);
+        r.assertLogContains("Perhaps you forgot to surround the code with a step that provides this, such as: sauce", run);
+    }
+}
```

Comment:
Moiz: Added braces

On Mon, Apr 11, 2016 at 10:30 AM, Moiz Virani <notifications@github.com>
wrote:

> In src/test/java/com/saucelabs/jenkins/pipeline/SauceStepTest.java
> <https://github.com/jenkinsci/sauce-ondemand-plugin/pull/15#discussion_r59245784>
> :
>
> > +    @Test
> > +    public void sauceConnectWithoutSauceTest() throws Exception {
> > +        DumbSlave s = r.createSlave();
> > +        WorkflowJob p = r.jenkins.createProject(WorkflowJob.class, "p");
> > +        p.setDefinition(new CpsFlowDefinition(
> > +            "node('" + s.getNodeName() + "') { sauceconnect(useGeneratedTunnelIdentifier: true, verboseLogging: true) { \n" +
> > +                "echo 'USERNAME=' + env.SAUCE_USERNAME\n" +
> > +                "echo 'ACCESS_KEY=' + env.SAUCE_ACCESS_KEY\n" +
> > +                "}}",
> > +            true
> > +        ));
> > +        WorkflowRun run = p.scheduleBuild2(0).get();
> > +        r.assertBuildStatus(Result.FAILURE, run);
> > +        r.assertLogContains("Perhaps you forgot to surround the code with a step that provides this, such as: sauce", run);
> > +    }
> > +}
>
> wow it disappeared!
> [image: screen shot 2016-04-11 at 10 29 24 am]
> <https://cloud.githubusercontent.com/assets/1308607/14436173/4de1e308-ffd0-11e5-8603-b81c85bf589d.png>
>
> I saw some thing like above
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/jenkinsci/sauce-ondemand-plugin/pull/15/files/ef414a4db9a6c8837062803b9e03f1f6e099f081#r59245784>
>



-- 


*Gavin MoganSoftware Developer*

Sauce Labs <http://saucelabs.com/> | Sauce blog <http://sauceio.com/>

Talk to us anytime:  <https://twitter.com/saucelabs>
<https://www.linkedin.com/company/sauce-labs>
<https://plus.google.com/+SauceLabs_Official/posts>
<https://www.facebook.com/saucelabs>